src/Heap.sol

```diff
@@ -36,209 +40,202 @@ library BasicHeap {
-
-        while (childRank <= size) {
-            // Compute the rank of the child with largest value.
-            if (
-                childRank < size &&
-                getAccount(_heap, childRank + 1).value > getAccount(_heap, childRank).value
-            ) childRank++;
-
-            childAccount = getAccount(_heap, childRank);
-
-            if (childAccount.value > initialValue) {
-                setAccount(_heap, _rank, childAccount);
-                _rank = childRank;
-                childRank *= 2;
-            } else break;
+    /// @notice Checks if an account with the given address `_id` exists in the `_heap`.
+    /// @dev The parameters `_index`, `_accountsLength`, and `_id` must be coherent.
+    /// @param _heap The heap to search in.
+    /// @param _index The index to search for the account in.
+    /// @param _accountsLength The length of the `_heap` accounts array.
+    /// @param _id The address of the account to search for.
+    /// @return True if the account exists in the `_heap`, false otherwise.
+    function containsAccount(Heap storage _heap, uint256 _index, uint256 _accountsLength, address _id)
+        private
+        view
+        returns (bool)
+    {
+        if (_index != 0) {
+            return true;
+        } else if (_accountsLength != 0) {
+            return _heap.accounts[0].id == _id;
+        } else {
+            return false;
         }
-        setAccount(_heap, _rank, initialAccount);
     }
 
     /// VIEW ///
 
-    /// @notice Returns the size of the `_heap`.
+    /// @notice Returns the number of users in the `_heap`.
     /// @param _heap The heap parameter.
-    /// @return The size of the heap.
+    /// @return The length of the heap.
     function getSize(Heap storage _heap) internal view returns (uint256) {
         return _heap.accounts.length;
     }
 
-    /// @notice Returns the value of the account linked to `_id`, returns 0 if the account is not in the `_heap`.
+    /// @notice Checks if the account with the given address `_id` exists in the `_heap`.
+    /// @param _heap The heap to search in.
+    /// @param _id The address of the account to search for.
+    /// @return True if the account exists in the heap, false otherwise.
+    function containsAccount(Heap storage _heap, address _id) internal view returns (bool) {
+        return containsAccount(_heap, _heap.indexOf[_id], _heap.accounts.length, _id);
+    }
+
+    /// @notice Returns the value of the account linked to `_id`.
     /// @param _heap The heap to search in.
     /// @param _id The address of the account.
     /// @return The value of the account.
     function getValueOf(Heap storage _heap, address _id) internal view returns (uint256) {
-        uint256 rank = _heap.ranks[_id];
-        if (rank == 0) return 0;
-        else return getAccount(_heap, rank).value;
+        uint256 index;
+
+        if (!containsAccount(_heap, index = _heap.indexOf[_id], _heap.accounts.length, _id)) return 0;
```

Comment:
```suggestion
        uint256 index = _heap.indexOf[_id];

        if (!containsAccount(_heap, index, _heap.accounts.length, _id)) return 0;
```
clearer IMO

Actually can we just return `_heap.accounts[index].value` and not do the `if` ?

Reply:
`accounts` is an array not a mapping, it would revert